File: test/formats/gfx/spb_converter_test.rb
require 'lib/formats/gfx/spb_converter'
require 'lib/virtual_file'
require 'test/test_helper'

# Unit tests for SpbConverter
class SpbConverterTest < Test::Unit::TestCase
  def test_decoding
    input_data = TestHelper.get_test_file('spb/grimoire_btn.bmp')
    expected_data = TestHelper.get_test_file('spb/grimoire_btn-out.png')

    file = VirtualFile.new(nil, input_data)
    SpbConverter.decode!(file, {})
    actual_data = file.data

    TestHelper.compare_image(expected_data, actual_data)
  end
end

File: tests/formats/gfx/spb_converter_test.cc
#include "formats/gfx/spb_converter.h"
#include "test_support/converter_support.h"

void test_spb_decoding(
    const std::string &input_path, const std::string expected_path)
{
    SpbConverter converter;
    assert_decoded_image(converter, input_path, expected_path);
}

int main(void)
{
    test_spb_decoding(
        "tests/test_files/gfx/spb/grimoire_btn.bmp",
        "tests/test_files/gfx/spb/grimoire_btn-out.png");
    return 0;
}
